### PR TITLE
feat(utility): add function to convert PyArrow schema to Avro schema

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-29qmwuq 添加 pyarrow schema 到 avro schema 的转换代码](https://app.clickup.com/t/29qmwuq)

File: graviti/utility/avro.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+
+# pylint: disable=missing-class-docstring
```
Comment (linjiX) [inline graviti/utility/avro.py:3]: can we disable all pylint warninig?

File: graviti/utility/avro.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    elif isinstance(_arrow_field, pa.DictionaryType):
+        # return _on_enum(names, namespace, name, _arrow_field)
+        return _on_primitive(_arrow_field.value_type)
+    elif isinstance(_arrow_field, pa.DataType):
+        return _on_primitive(_arrow_field)
+    else:
+        raise f"unsupported type {_arrow_field}"
+
+
+def convert_arrow_schema_to_avro(_schema: pa.Schema):
```
Comment (linjiX) [inline graviti/utility/avro.py:230]: shall we expose this method in `__init__.py`?

File: graviti/portex/field.py [hunk 3]
```diff
@@ -142,7 +142,10 @@ def to_pyarrow(self) -> List[pa.Field]:
             A list of PyArrow Field representing the fields of Portex record.
 
         """
-        return [pa.field(name, portex_type.to_pyarrow()) for name, portex_type in self.items()]
+        return [
+            pa.field(name, portex_type.to_pyarrow(), nullable=False)
```
Comment (linjiX) [inline graviti/portex/field.py:146]: should we get the `nullable` from `Portex.nullable`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/avro.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+
+# pylint: disable-all
+# type: ignore[no-untyped-def]
+# flake8: noqa: D100, D101, D102, D103
```
Comment (linjiX) [inline graviti/utility/avro.py:5]: disable all flake8 error?

File: graviti/portex/builtin.py [hunk 5]
```diff
@@ -397,7 +399,7 @@ def to_pyarrow(self) -> pa.DataType:
             if len(pytypes) == 1
             else pa.union(
                 (
-                    pa.field(str(index), _PYTHON_TYPE_TO_PYARROW_TYPE[pytype])
+                    pa.field(str(index), _PYTHON_TYPE_TO_PYARROW_TYPE[pytype], nullable=False)
```
Comment (linjiX) [inline graviti/portex/builtin.py:402]: what if there is a "None" in the enum.values?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**